src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -567,7 +598,7 @@ protected virtual string GenerateDisplayName([NotNull] ModelMetadata metadata, s
             if (resolvedDisplayName == null)
             {
                 resolvedDisplayName =
-                    string.IsNullOrEmpty(htmlFieldName) ? string.Empty : htmlFieldName.Split('.').Last();
+                    string.IsNullOrEmpty(expression) ? string.Empty : expression.Split('.').Last();
```

Comment:
Curious: Saw this earlier in the PR too, so we don't protect against `.Last` throwing in our HTML helpers? Or does the caller catch the potential throw cases?